pkg/frontend/v2/frontend.go

```diff
@@ -207,17 +207,16 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 	}
 
 	spanLogger := spanlogger.FromContext(ctx, f.log)
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	ctx, cancel := context.WithCancelCause(ctx)
+	defer cancel(cancellation.NewErrorf("Frontend.RoundTripGRPC finished"))
```

Comment:
given that those can be shown to users and operators should we try to make them a little less cryptic? Maybe something like "executing subquery finished"?

Reply:
Why would this be shown to the user? Aren't we deferring just to avoid leaked resources?

Reply:
the reason we're adding these causes is because if they do reach logs or user responses we can track them down. My take was that making them understandable without having to read the code is a benefit. It's easier to grok even if you're a maintainer/operator, not necessarily a user

Reply:
Since they're not really expected to reach the user or even logs, I agree with any contents as long as they're unique. 

IMO the method name here is useful to quickly understand where it comes from.

Reply:
I've changed this to `executing query round trip finished` in db107b9.